src/main/java/mods/railcraft/common/blocks/tracks/instances/TrackSwitchBase.java

```diff
@@ -164,23 +165,19 @@ public void onBlockRemoved() {
     }
 
     protected void determineRailDirection() {
-        IBlockState current = getWorld().getBlockState(getPos());
-        IProperty<EnumRailDirection> shapeProp = ((BlockRailBase) current.getBlock()).getShapeProperty();
-        EnumRailDirection dir = current.getValue(shapeProp);
+        EnumRailDirection dir = TrackTools.getTrackDirection(getWorld(), getPos());
```

Comment:
Its shouldn't, but I'll change it to raw anyway as it might cause problems later.